### PR TITLE
CIVDT-828: User Type page navigation and data cleanup on changing response

### PR DESCRIPTION
**Pull request by macgrewal**

### CODE REVIEW

File: app/controllers/UserTypeController.scala [hunk 1]
```diff
@@ -41,7 +42,20 @@ class UserTypeController @Inject()(identify: IdentifierAction,
...
+        mode <- answers.get(CheckModePage)
+      } yield mode
+    }.getOrElse(false)
+
+    if (cyaMode) {
+      controllers.routes.CheckYourAnswersController.onLoad()
+    } else {
+      controllers.routes.ConfirmEORIDetailsController.onLoad()
+    }
+  }
```
Comment (dannybassi) [inline app/controllers/UserTypeController.scala:58]: We would normally do the back links in the below format. Is there is reason to why you done the back link like you have? 

 private[controllers] def backLink()(implicit request: DataRequest[_]): Call = {
    if (request.checkMode) {
      controllers.routes.CheckYourAnswersController.onLoad()
    } else {
      controllers.routes.ConfirmEORIDetailsController.onLoad()
    }
  }

File: app/viewmodels/cya/CYAYourDetailsSummaryListHelper.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ trait CYAYourDetailsSummaryListHelper {
       createRow(
         keyText = Text(messages("cya.userType")),
         valueContent = Text(userTypeValue),
-        action = Some(ActionItem("Url", Text(messages("cya.change"))))
+        action = Some(ActionItem(controllers.routes.UserTypeController.onLoad().url, Text(messages("cya.change"))))
```
Comment (dannybassi) [inline app/viewmodels/cya/CYAYourDetailsSummaryListHelper.scala:60]:
```suggestion
        action = Some(ActionItemHelper.createChangeActionItem(
          controllers.routes.UserTypeController.onLoad().url,
          messages("cya.userType.change")
        ))
```

File: test/views/data/CheckYourAnswersData.scala [hunk 3]
```diff
@@ -259,7 +259,7 @@ object CheckYourAnswersData {
           ),
           actions = Some(Actions(items = Seq(
             ActionItem(
-              changeUrl,
+              controllers.routes.UserTypeController.onLoad().url,
               Text(CYAMessages.change))),
```
Comment (dannybassi) [inline test/views/data/CheckYourAnswersData.scala:263]:
```suggestion
            ActionItem(
              controllers.routes.UserType.onLoad().url,
              HtmlContent("""<span aria-hidden="true">Change</span>"""),
              visuallyHiddenText = Some(CYAMessages.changeUserType)
            )
              )),
```

**Review by dannybassi — CHANGES_REQUESTED** [review]
Few changes

File: app/controllers/UserTypeController.scala (hunk 1 above)
Comment (macgrewal) [reply app/controllers/UserTypeController.scala:58]: Yes, typical controllers include the `DataRequiredAction` that attaches certain functions to the `request` such as `request.checkMode`. However, the `UserTypeController` only has the `DataRetrievalAction` which doesn't have the associated helper functions so I've replicated the functiobnality for this instance.

File: app/viewmodels/cya/CYAYourDetailsSummaryListHelper.scala (hunk 2 above)
Comment (macgrewal) [reply app/viewmodels/cya/CYAYourDetailsSummaryListHelper.scala:60]: Changed and updated the visually hidden message to include some context around the change action.

File: test/views/data/CheckYourAnswersData.scala (hunk 3 above)
Comment (macgrewal) [reply test/views/data/CheckYourAnswersData.scala:263]: Done

**Pull request merged**